File: module-viewbinding/src/main/java/com/chiatai/module_viewbinding/MainActivity.java
package com.chiatai.module_viewbinding;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.chiatai.module_viewbinding.databinding.ActivityMainBinding;

/**
 * ViewBinding 存在的目的就是为了 不用写findViewById()
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        // binding.getRoot()是获取到布局文件中根布局的实例
        setContentView(binding.getRoot());
        binding.tv.setText("您好。。。。");
    }
}

File: module-databinding/src/main/java/com/chiatai/module/databinding/MainActivity.kt
package com.chiatai.module.databinding

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.chiatai.module.databinding.databinding.ActivityMainBinding
import com.chiatai.module.databinding.inverse.InverseViewModel

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        var activityBinding =
            DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        activityBinding.lifecycleOwner = this
        val inverseViewModel = ViewModelProvider(this).get(InverseViewModel::class.java)
        activityBinding.viewModel = inverseViewModel

        inverseViewModel.inverseBean.index.observe(this, Observer {
            it?.let {
                Log.e("inverse", "index == $it")
            }
        })

    }
}
